MAUI/DataForm/Editors.md

```diff
@@ -0,0 +1,1187 @@
+
+{% tabs %}
+{% highlight C# %}
+
+    public string Country { get ; set ; }
+    public class DataFormItemsSourceProvider : IDataFormSourceProvider
+    {
+         public object GetSource(string sourceName)
+        {
+            if (sourceName == "Country")
+            {
+                List<string> list = new List<string>()
+                {
+                    "Indonesia",
+                    "Italy",
+                    "India",
+                    "Iran",
+                    "Iraq",
+                    "Uganda"
+                    "Ukraine"
+                    "Canada"
+                    "Australia"
+                    "Uzbekistan"
+                    "France"
+                    "United Kingdom"
+                    "United States"
+                };
+                return list;
+            }
+            return new List<string>();
+        }
+    }
+    dataForm.ItemsSourceProvider = new DataFormItemsSourceProvider();
+    dataForm.RegisterEditor("Country", DataFormEditorType.AutoComplete);
+
+{% endhighlight %}
+{% endtabs %}
+
+`Image`
+
+#### Using GenerateDataFormItem event
+
+You can also set `ItemsSource` for autocomplete editor by using `ItemsSource` property in the `DataFormAutoCompleteItem`.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.RegisterEditor("Country", DataFormEditorType.AutoComplete);
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null && e.DataFormItem.FieldName == "Country" && e.DataFormItem is DataFormAutoCompleteItem autoCompleteItem)
+        {
+            List<string> list = new List<string>();
+            list.Add("Indonesia");
+            list.Add("Italy");
+            list.Add("India");
+            list.Add("Iran");
+            list.Add("Iraq");
+            list.Add("Uganda");
+            list.Add("Ukraine");
+            list.Add("Canada");
+            list.Add("Australia");
+            list.Add("Uzbekistan");
+            list.Add("France");
+            list.Add("United Kingdom");
+            list.Add("United States");
+            autoCompleteItem.ItemsSource = list;
+        }
+    }
+{% endhighlight %}
+{% endtabs %}
+
+#### Loading the complex type property values in autocomplete editor
+
+You can display the complex type property values in combo box editor by using the DisplyMemberPath and SelectedValuePath properties of DataFormComboBoxItem. You need to use `GenerateDataFormItem `event to set `DisplayMemberPath` and `SelectedValuePath` property value of DataFormComboBoxItem for complex type property.
+
+You can display the complex type property values in autocomplete editor by using the DisplayMemberPath and SelectedValuePath properties of DataFormAutoCompleteItem. Use the `GenerateDataFormItem `event to set `DisplayMemberPath` and `SelectedValuePath` property values of AutoComplete for complex type property.
```

Comment:
remove You can display the complex type property values in autocomplete editor by using the DisplayMemberPath and SelectedValuePath properties of DataFormAutoCompleteItem. Use the `GenerateDataFormItem `event to set `DisplayMemberPath` and `SelectedValuePath` property values of AutoComplete for complex type property.

Reply:
above mentioned content removed